x/qgb/README.md

```diff
@@ -1,50 +1,320 @@
+// It does not contain a `commitment` field as this message will be created
+// inside the state machine and it doesn't make sense to ask tendermint for the
+// commitment there.
+// The range defined by begin_block and end_block is end exclusive.
+message DataCommitment {
+  option (cosmos_proto.implements_interface) = "AttestationRequestI";
+  // Universal nonce defined under:
+  // https://github.com/celestiaorg/celestia-app/pull/464
+  uint64 nonce = 1;
+  // First block defining the ordered set of blocks used to create the
+  // commitment.
+  uint64 begin_block = 2;
+  // End exclusive last block defining the ordered set of blocks used to create
+  // the commitment.
+  uint64 end_block = 3;
+  // Block time where this data commitment was created
+  google.protobuf.Timestamp time = 4
+      [ (gogoproto.nullable) = false, (gogoproto.stdtime) = true ];
+}
+```
+
+1. `nonce`: is the universal nonce of the attestation. It is used to prevent front running attacks in the QGB smart contract. More details can be found in [ADR-004](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-004-qgb-relayer-security.md).
+2. `begin_block`: the data commitment range first block.
+3. `end_block`: the data commitment range last block. The range is end exclusive. Thus, the commitment will be over the set of blocks defined by `[begin_block, end_block)`.
+4. `time`: is the timestamp of the height at which the data commitment was created.
+
+A commitment, aka [`DataRootTupleRoot`](https://github.com/celestiaorg/quantum-gravity-bridge/blob/03f0ab6f6eceba7a7c74a46b2c2a1fd1e390010c/src/QuantumGravityBridge.sol#L165-L180), is an RFC-6962 [merkle tree commitment](https://github.com/celestiaorg/celestia-core/blob/d280f37a8376ed54ae03b10896fa25a4cbbc6d5b/rpc/core/blocks.go#L344-L361) over the set of [`DataRootTuples`](https://github.com/celestiaorg/quantum-gravity-bridge/blob/03f0ab6f6eceba7a7c74a46b2c2a1fd1e390010c/src/DataRootTuple.sol#L4-L17) defined by a data commitment range.
+
+A data root tuple contains the following fields:
+
+```solidity
+/// @notice A tuple of data root with metadata. Each data root is associated
+///  with a Celestia block height.
+/// @dev `availableDataRoot` in
+///  https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/data_structures.md#header
+struct DataRootTuple {
+    // Celestia block height the data root was included in.
+    // Genesis block is height = 0.
+    // First queryable block is height = 1.
+    uint256 height;
+    // Data root.
+    bytes32 dataRoot;
+    // Celestia block original square size.
+    uint256 squareSize;
 }
 ```
 
-I.e. the current block height is not 0, and we're at a data commitment window height.
+1. `height`: the height of the block.
+2. `dataRoot`: the data root, aka data hash, of the block.
+3. `squareSize`: the [square](https://celestiaorg.github.io/celestia-app/specs/data_structures.html#arranging-available-data-into-shares) size of the block.
```

Comment:
Why is the square size important?

Reply:
I didn't want to add it because there isn't any write-up for it so far.  I will open an issue to add it

Reply:
https://github.com/celestiaorg/celestia-app/issues/1918